app/src/main/java/br/org/cesar/discordtime/stickysessions/presentation/session/SessionPresenter.java

```diff
@@ -14,39 +16,94 @@
     private DisposableSingleObserver<List<Note>> mListNotesObserver;
+    private DisposableSingleObserver<String> mGetSaveUserObserver;
+    private DisposableSingleObserver<Boolean> mSaveCurrentUserObserver;
+    private SessionContract.View mView;
+    private Session mActiveSession;
+    private String mSessionId;
+    private String mCurrentUser;
+    private final List<DisposableSingleObserver> mDisposableObjects;
 
     public SessionPresenter(ObservableUseCase<String, Session> enterSession,
                             ObservableUseCase<Note, Note> addNote,
-                            ObservableUseCase<String, List<Note>> listNotes) {
+                            ObservableUseCase<NoteFilter, List<Note>> listNotes,
+                            ObservableUseCase<String, Boolean> saveCurrentUser,
+                            ObservableUseCase<Void, String> getSavedUser) {
         mEnterSession = enterSession;
         mAddNote = addNote;
         mListNotes = listNotes;
+        mSaveCurrentUser = saveCurrentUser;
+        mGetSavedUser = getSavedUser;
+
+        mDisposableObjects = new ArrayList<>();
     }
 
     @Override
     public void attachView(SessionContract.View view) {
         mView = view;
     }
 
+    @Override
+    public void onResume() {
+        mGetSaveUserObserver = new DisposableSingleObserver<String>() {
+            @Override
+            public void onSuccess(String userName) {
+                mCurrentUser = userName;
+                onEnterSession();
+            }
+
+            @Override
+            public void onError(Throwable e) {
+                mView.showWidgetAddName();
+            }
+        };
+        mGetSavedUser.execute(mGetSaveUserObserver, null);
+        mDisposableObjects.add(mGetSaveUserObserver);
+    }
+
+    @Override
+    public void currentUser(String userName) {
+        mSaveCurrentUserObserver = new DisposableSingleObserver<Boolean>() {
+            @Override
+            public void onSuccess(Boolean success) {
+                if (success) {
+                    mCurrentUser = userName;
+                    onEnterSession();
+                }
+            }
+
+            @Override
+            public void onError(Throwable e) {
+                mView.showWidgetAddName();
+                mView.displayError(e.getMessage());
+            }
+        };
+        mSaveCurrentUser.execute(mSaveCurrentUserObserver, userName);
+        mDisposableObjects.add(mSaveCurrentUserObserver);
+    }
+
+    @Override
+    public void currentSession(String sessionId) {
+        mSessionId = sessionId;
+    }
+
     @Override
     public void detachView() {
         mView = null;
-        if (mSessionObserver != null && !mSessionObserver.isDisposed()) {
-            mSessionObserver.dispose();
-        }
-
-        if (mNoteObserver != null && !mNoteObserver.isDisposed()) {
-            mNoteObserver.dispose();
-        }
+        disposeObservers();
```

Comment:
We could use the already implemented `ObservableUseCase.dispose()` method from each UseCase instead of storing disposables again in a list.

https://github.com/DiscordTime/sticky-sessions-android/blob/dev/app/src/main/java/br/org/cesar/discordtime/stickysessions/executor/ObservableUseCase.java#L35